_posts/en/newsletters/2019-02-19-newsletter.md

```diff
@@ -0,0 +1,252 @@
+  reflected in the long list of changes in this week's *notable changes*
+  section below.  The next two weeks will focus on developer testing and
+  bug fixes, followed by the issuance of Release Candidates (RCs) for
+  user testing.  The RC cycle for a major release usually lasts two to
+  four weeks before a final release.
+
+    Related, the project prefers to merge major new features early in a
+    new development cycle so that they get as much post-ACK developer
+    testing as possible.  After the 0.18 branch is created around March
+    1st, anyone who wants to see a feature in 0.19 (estimated release
+    October 2019) would be advised to either try to open a PR for it
+    within the next two months or to assist in reviewing an existing PR
+    for that feature.  Some notable existing PRs that need more review
+    or development include support for BIP156 Dandelion
+    [privacy-enhanced transaction relay][Bitcoin Core #13947], BIP151
+    [encrypted P2P connections][Bitcoin Core #14032], BIP157/158
+    [compact block filters][Bitcoin Core #14212], simplified
+    [reproducible builds][Bitcoin Core #15277] using GNU Guix, improved
+    support for [external signers][Bitcoin Core #14912] (e.g. hardware
+    wallets), [separating the wallet from the node][Bitcoin Core
+    #10973], and allowing [RBF on any transaction][Bitcoin Core #10823]
+    after it's been in the mempool for more than a few hours.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[LND][lnd repo], [C-Lightning][c-lightning repo], [Eclair][eclair repo],
+[libsecp256k1][libsecp256k1 repo], and [Bitcoin Improvement Proposals
+(BIPs)][bips repo].*
+
+- [Bitcoin Core #15368][] adds support for checksums to output script
+  descriptors.  [Descriptors][descriptor] are used to monitor for
+  received payments and generate new addresses, so checksums improve
+  safety by preventing copying errors that could cause money to go
+  missing or be sent to an unspendable address.  A `#` character is
+  added to the descriptor grammar for separating the descriptor from its
+  8-character checksum, e.g.  `wpkh(031234...cdef)#012345678` (see
+  footnote[^fn-example] for an extended example). All Bitcoin Core RPCs
+  that return descriptors now include a checksum.  RPCs that aren't
+  particularly at risk of losing money don't require that input include
+  the checksum, but RPCs that are safety critical, such as
+  `deriveaddress` and `importmulti`, are updated to require users
+  provide the checksum.  Finally, a new `getdescriptorinfo` RPC is added
+  that accepts a descriptor and returns a normalized form of it
+  containing a checksum along with some other information about it.
+
+- [Bitcoin Core #13932][] adds three new RPCs for managing PSBTs:
+  `utxoupdatepsbt` uses information in the UTXO set to determine whether
+  any inputs in the PSBT use native segwit and flags them accordingly.
```

Comment:
I don't think this is quite right. 'Flags them accordingly' suggests to me that the command simply says "this is a segwit UTXO". I think what it's actually doing is finding the UTXO in the UTXO set and inserting it into the PSBT.

Reply:
You're correct; I confused this with another PSBT change a few weeks ago related to disentangling witnessScript and redeemScript by simply flagging which was what.   A problem I do have with this PR (and also my earlier description and your correction of it) is that no motivation is provided for why someone would use this RPC.  The other two RPCs in this PR are named such that the motivation for using them is clear, but "utxo update" makes no sense---it's not like UTXOs change or that this RPC can automatically take an old outpoint for a tx no longer in the chain and provide a new outpoint whose output is otherwise identical.

So I extended this description to three sentences.  The first two describe how it works based on my revised understanding; the third offers a motivation for using it that I've hypothesized out of my own imagination.  I'd like to include that motivation (or any motivation!) in the newsletter as possible to help people understand why they might use this feature, but if you find the hypothesis weak, clearly wrong, or simply don't want us to speculate, I suggest you simply delete that third sentence at merge time.

Reply:
I think your motivation is exactly correct. I could hand my hardware wallet an unsigned bitcoin transaction and say 'sign this'. If any of the inputs to that transaction were segwit outputs, the hardware wallet would not be able to sign the transaction because the signature covers parts of the UTXO being spent. `utxoupdatepsbt` allows me to put my unsigned transaction into a PSBT and ask bitcoind to populate the required `Witness UTXO` fields.

Reply:
I've removed *version 0* from your description. Although technically true, I think *version 0* probably doesn't mean anything to most of our readers and is more confusing than clarifying at this point.